packages/dataviews/tsconfig.json

```diff
@@ -0,0 +1,20 @@
+{
```

Comment:
According to https://github.com/WordPress/gutenberg/blob/add/typescript-dataviews/packages/README.md?plain=1#L196 this looks fine.

Reply:
Came here to suggest that we should add a `types: "build-types"` in dataviews' `package.json` file, but it turns out we already have it. Not sure what whether there's any side-effects this could have (not having types vs having types after this PR), but thought worth mentioning it.

Reply:
No important impact aside that third-party devs using the package will now see some types appearing (autocompletion and all) when they use the package. Previously the build-types folder was empty, now it has the types defined in this PR.

Reply:
The `@wordpress/dataviews/build-types/` folder has the proper data (types + TS files).

Though I'm a bit confused. I wanted to see what we had previously, so I went to trunk and did `npm install && npm run build`. Prior to that, I removed everything (tried in a few different ways: manually `rm -rf node_modules`, as well as other options `npm run distclean` or `npm run clean:*`). The `@wordpress/dataviews/build-types` folder contained the types and TS files from this PR 😕 I presume there's a cache somewhere I'm not clearing. It'd be nice to learn how to clear this. Other than that, I suspect it's working as expected.

Reply:
You're right I have the same issue. I think the problem is that "dataviews" is not on the root tsconfig.json file in trunk which means when running `npm run clean:package-types`, typescript doesn't touch it at all, it's not aware of its existence so it's not cleared.